crates/gossip/src/timeout.rs

```diff
@@ -59,6 +69,14 @@ async fn remove_timed_out_rounds(
                     );
                 }
 
+                // If we failed to initiate, then we want to try again. Otherwise, an unavailable
+                // peer can prevent us from initiating gossip.
+                if matches!(state.stage, RoundStage::Initiated(_)) {
```

Comment:
On a timeout in the initiate stage, trigger initiate to run again. I think that's the only time we should do it on a timeout.